package/tvheadend-app-osmc/files/DEBIAN/postinst

```diff
@@ -1,33 +1,182 @@
-#!/bin/bash
+#! /bin/bash
 
 if [ "$1" = "configure" ]; then
 
 	systemctl daemon-reload
+#set -x
+	# going to read existing config files even if installing
+	# unless version is 4.2 when we can trust tvh to do it
+	EXISTINGVER=$(egrep "full_?version" /home/osmc/.hts/tvheadend/config | sed 's/.*: \+\"\(.*\)\".*/\1/' 2> /dev/null)
+	
+	if $(dpkg --compare-versions "$2" lt "4.2.0") || [[ "$EXISTINGVER" < "4.2.0" ]] \
+		|| [ -e "/var/run/${DPKG_MAINTSCRIPT_PACKAGE}_install" ]; then
+#	if ((1)); then
+		# check for existing user called osmc
+		if OSMCUSERFILES=$(ls -t /home/osmc/.hts/tvheadend/accesscontrol/* 2> /dev/null); then
```

Comment:
No.  The whole point is we need to know whether there are old config files which need to be fettled before tvh does its migration.  Those files can exist even if the OS thinks there wasn't a previous installation.